code/game/objects/structures/railing.dm

```diff
@@ -290,3 +299,10 @@
 	if(.)
 		if(!anchored || material.is_brittle())
 			take_damage(maxhealth) // Fatboy
+
+	var/start_layer = user.layer
+	user.layer = VEHICLE_LOAD_LAYER
+	addtimer(CALLBACK(src, .proc/reset_layer, start_layer, user), 2)
```

Comment:
Not a big fan of this implementation; do not mess with a mob's layering externally. If you want to do this, add logic on mobs that adjusts their layers.

Reply:
Uh, not sure what you mean, something like a mob proc that gets called when layers need changing?

Reply:
The current model mobs have to handle layering is: `reset_layer` sets the layer to whatever it should be, depending on the mob's state. This is called every `Life` tick, and can be called more frequently if need be. Setting the layer on the mob directly conflicts with this: it will be unset randomly the next time `Life` is called (so in 0-2 seconds) and at other times.

So, good implementation: set a state var on mobs (I think there even already are some, having to do with climbing), have the mob's `reset_layer` proc set layer to whichever one you use here if that state var is true, then call `reset_layer` on the mob here. In the callback, unset the state var and call `reset_layer` again.

Tolerable implementation comparable to mediocre code elsewhere: change the layer here, call `reset_layer` on the mob in the callback, do not cache the old layer.

Reply:
I'll also note that vehicles are not a good example: they do not actually change a mob's layer, despite how the code there looks like.

Reply:
Yes, just make sure multiple sources of layering changes are handled reasonably (first wins, last wins, whatever, just so long as it's consistent at dealing with the callbacks). You probably should block `reset_layer` and/or `reset_plane_and_layer`, not `UpdateLyingBuckledAndVerbStatus` (which, notably, handles pixel shifting, which is similarly brittle, has identical issues, and comes up a lot more).

Reply:
Alright, all done, thanks for the help. I guess we just wait for eckff now.
Probably a good idea to port stuff after they get merged, though